### PR TITLE
Support for using existing / requiring existing resource group(s).  

### LINKED ISSUE #474: Bedrock should allow for BYO resource group
**As a:** SRE

**I want:** to be able to bring my own resource group 

**So that:** Terraform can use my resource group and not delete it upon a Terraform delete.

**Does this require updates to documentation?:** Yes, there will be a new boolean in the terraform variables file for each environment such that one can toggle on/off resource group management

### PR DESCRIPTION
**Pull request by jmspring**
Closes #474

### CODE REVIEW

File: test/bedrock_Azure_simple_test.go [hunk 1]
```diff
@@ -34,6 +37,22 @@ func TestIT_Bedrock_AzureSimple_Test(t *testing.T) {
...
 
+	//Create the resource group
+	cmd0 := exec.Command("az", "login", "--service-principal", "-u", clientid, "-p", clientsecret, "--tenant", tenantId)
+	err0 := cmd0.Run()
+	if err0 != nil {
+		fmt.Println("unable to login to azure cli")
+		log.Fatal(err0)
+		os.Exit(-1)
+	}
+	cmd1 := exec.Command("az", "group", "create", "-n", k8sRG, "-l", location)
```
Comment (timfpark) [inline test/bedrock_Azure_simple_test.go]: Consider a follow up PR at some point to perhaps pull some of these common fixture setups into a library.

File: cluster/azure/README.md [hunk 2]
```diff
@@ -23,10 +23,11 @@ Beyond these, you'll only need the Azure `az` command line tool installed (used
 
 Bedrock provides different templates to start from when building your deployment environment.  Each template has a set of common and specific requirements that must be met in order to deploy them.
 
+Common across templates, it is required that the resource group(s) needed by the enviornment be created prior to deploying.  For how to create a resource group, see [here]()
+
```
[the remark this answers is not in the document]
Comment (jmspring) [reply cluster/azure/README.md]: fixed

File: cluster/common/velero/main.tf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module "common-provider" {
-  source = "github.com/microsoft/bedrock?ref=master//cluster/common/provider"
+  source = "github.com/microsoft/bedrock?ref=byo.rg//cluster/common/provider"
```
Comment (NathanielRose) [inline cluster/common/velero/main.tf]: Will we need to switch this back to master as the source?

File: test/bedrock_Azure_mc_test.go [hunk 4]
```diff
@@ -61,6 +64,22 @@ func TestIT_Bedrock_AzureMC_Test(t *testing.T) {
 	azureCommonInfraFolder := "../cluster/test-temp-envs/azure-common-infra-" + k8sName
 	copy.Copy("../cluster/environments/azure-common-infra", azureCommonInfraFolder)
 
+        //Create the common resource group
+        cmd0 := exec.Command("az", "login", "--service-principal", "-u", clientid, "-p", clientsecret, "--tenant", tenantid)
```
Comment (NathanielRose) [inline test/bedrock_Azure_mc_test.go]: In the future, we may be able to use the Terratest module to create the RG.  - https://github.com/gruntwork-io/terratest/pull/333

File: cluster/common/velero/main.tf (hunk 3 above)
Comment (jmspring) [reply cluster/common/velero/main.tf]: Yes

File: test/bedrock_Azure_mc_test.go (hunk 4 above)
Comment (jmspring) [reply test/bedrock_Azure_mc_test.go]: That's a good fix.  Let's file an issue on it.

Comment (NathanielRose) [reply test/bedrock_Azure_mc_test.go]: Issue for reference - https://github.com/microsoft/bedrock/issues/553

**Pull request merged**